### PR TITLE
Jcf map, set / it-academy / JC1-110-22 / Evgeniy Antonovich / Done tasks Map and Set

### PR DESCRIPTION
**Pull request by aevgeni**
## Pull request state

Actual status for PR:
- [x] implemented
- [x] reviewed
- [x] fixed
- [ ] approved

### CODE REVIEW

File: src/main/java/com/rakovets/course/java/core/practice/jcf_set/Citizen.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
+package com.rakovets.course.java.core.practice.jcf_set;
+
+public class Citizen implements Comparable<Citizen> {
+    private String firstName;
+    private String lastName;
+    private String locationCity;
+    private final int yearOfBirth;
+    private final int id;
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/jcf_set/Citizen.java:8]: fix
```suggestion
    private final int id;
    private final int yearOfBirth;
    private String firstName;
    private String lastName;
    private String locationCity;
```

File: src/main/java/com/rakovets/course/java/core/practice/jcf_set/Fine.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
+package com.rakovets.course.java.core.practice.jcf_set;
+
+public class Fine implements Comparable<Fine> {
+    private double articleNumber;
+    private String description;
+    private double fineAmount;
+    private boolean isPaid;
+    private final int citizenId;
+    private final int fineId;
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/jcf_set/Fine.java:9]: fix
```suggestion
    private final int citizenId;
    private final int fineId;
    private double articleNumber;
    private String description;
    private double fineAmount;
    private boolean isPaid;
```

File: src/main/java/com/rakovets/course/java/core/practice/jcf_set/TaxService.java [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+        this.fines = fines;
+    }
+
+    public Citizen getCitizenById(int citizenId) {
+        for (Citizen citizen : citizens) {
+            if (citizen.getId() == citizenId) {
+                return citizen;
+            }
+        }
+        return null;
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/jcf_set/TaxService.java:23]: throw Exception

File: src/main/java/com/rakovets/course/java/core/practice/jcf_set/TaxService.java [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+        return finesByCitizenId;
+    }
+
+    public Fine getFineById(int fineId) {
+        for (Fine fine : fines) {
+            if (fine.getFineId() == fineId) {
+                return fine;
+            }
+        }
+        return null;
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/jcf_set/TaxService.java:62]: throw Exception

File: src/main/java/com/rakovets/course/java/core/practice/jcf_set/TaxService.java [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+
+    public void removeFineById(int fineId) {
+        Iterator<Fine> iterator = fines.iterator();
+        while (iterator.hasNext()) {
+            if (iterator.next().getFineId() == fineId) {
+                iterator.remove();
+                break;
+            }
+        }
+    }
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/jcf_set/TaxService.java:73]: Case: `fineId` don't exists

File: src/test/java/com/rakovets/course/java/core/practice/jcf_set/TaxServiceDemo.java [hunk 6]
```diff
@@ -0,0 +1,46 @@
+package com.rakovets.course.java.core.practice.jcf_set;
+
+import java.util.*;
```
Comment (rakovets) [inline src/test/java/com/rakovets/course/java/core/practice/jcf_set/TaxServiceDemo.java:3]: Don't use * import

**Review by rakovets — CHANGES_REQUESTED** [review]
Need to fix

File: src/main/java/com/rakovets/course/java/core/practice/jcf_set/TaxService.java (hunk 3 above)
Comment (aevgeni) [reply src/main/java/com/rakovets/course/java/core/practice/jcf_set/TaxService.java]: did it

File: src/main/java/com/rakovets/course/java/core/practice/jcf_set/TaxService.java (hunk 4 above)
Comment (aevgeni) [reply src/main/java/com/rakovets/course/java/core/practice/jcf_set/TaxService.java]: did it

File: src/main/java/com/rakovets/course/java/core/practice/jcf_set/TaxService.java (hunk 5 above)
Comment (aevgeni) [reply src/main/java/com/rakovets/course/java/core/practice/jcf_set/TaxService.java]: added "throw Exception"

File: src/test/java/com/rakovets/course/java/core/practice/jcf_set/TaxServiceDemo.java (hunk 6 above)
Comment (aevgeni) [reply src/test/java/com/rakovets/course/java/core/practice/jcf_set/TaxServiceDemo.java]: fixed it

**Pull request closed without merging**